src/EventBand/Transport/AmqpLib/Pool/Strategy/StrategyFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * @LICENSE_TEXT
+ */
+
+namespace EventBand\Transport\AmqpLib\Pool\Strategy;
+
+/**
+ * Class StrategyFactory
+ * @package EventBand\Transport\AmqpLib\Strategy
+ */
+class StrategyFactory
+{
+    /**
+     * @param string $strategyName
+     * @return StrategyInterface
+     */
+    public function create($strategyName)
+    {
+        $strategy = null;
+
+        $strategyClass = '\\EventBand\\Transport\\AmqpLib\\Pool\\Strategy\\Strategy' . $strategyName;
+        if (class_exists($strategyClass)) {
+            $strategy = new $strategyClass();
+        }
+
+        if (!$strategy instanceof StrategyInterface) {
+            throw new \InvalidArgumentException('Can\'t load strategy class');
+        }
+
+        return $strategy;
+    }
+}
```

Comment:
This factory has so many problems:
1) You can not create strategies with non-empty constructors
2) It will be hard for refactoring with hard coded namespace
3) It is static by the nature, but we still have to create object

This is a bad pattern for pure code, such logic can be in bundles only.